server/src/main/java/org/elasticsearch/plugins/PluginsLoader.java

```diff
@@ -64,18 +64,26 @@ public interface PluginLayer {
          * @return The {@link ClassLoader} used to instantiate the main class for the plugin
          */
         ClassLoader pluginClassLoader();
+
+        /**
+         * @return The {@link ModuleLayer} for the plugin modules
+         */
+        ModuleLayer pluginModuleLayer();
     }
 
     /**
      * Contains information about the {@link ClassLoader}s and {@link ModuleLayer} required for loading a plugin
-     * @param pluginBundle Information about the bundle of jars used in this plugin
+     *
+     * @param pluginBundle      Information about the bundle of jars used in this plugin
      * @param pluginClassLoader The {@link ClassLoader} used to instantiate the main class for the plugin
-     * @param spiClassLoader The exported {@link ClassLoader} visible to other Java modules
-     * @param spiModuleLayer The exported {@link ModuleLayer} visible to other Java modules
+     * @param pluginModuleLayer The {@link ModuleLayer} containing the Java modules of the plugin
+     * @param spiClassLoader    The exported {@link ClassLoader} visible to other Java modules
+     * @param spiModuleLayer    The exported {@link ModuleLayer} visible to other Java modules
      */
     private record LoadedPluginLayer(
         PluginBundle pluginBundle,
         ClassLoader pluginClassLoader,
+        ModuleLayer pluginModuleLayer,
```

Comment:
Fear not, this is not added here just for this test: we need it anyway/it's one of the reasons we wanted to split PluginsLoader out (see it in use in https://github.com/elastic/elasticsearch/pull/117239)